plugins/parsers/registry.go

```diff
@@ -145,6 +161,28 @@ func NewParser(config *Config) (Parser, error) {
 	return parser, err
 }
 
+func newJSONParser(
+	metricName string,
+	tagKeys []string,
+	stringFields []string,
+	jsonQuery string,
+	timeKey string,
+	timeFormat string,
+	defaultTags map[string]string,
+) (Parser, error) {
```

Comment:
Returning an error isn't useful here. Where it's an unexported function not matching any interface, I'd remove it, or remove the function altogether and just instantiate a `JSONParser` on line 120

Reply:
No error is ever returned by the new_Parser functions, but i thought i'd include it here to match the format of the rest of the parsers